app/src/main/java/com/example/fishkeeper/network/CatchPost.kt

```diff
@@ -3,12 +3,12 @@ package com.example.fishkeeper.network
 import com.google.gson.annotations.SerializedName
 
 data class CatchPost(
-    val latitude: Double,
-    val longitude: Double,
-    val altitude: Double,
-    @SerializedName("fish_length") val fishLength: Double,
+    val latitude: Double?,
+    val longitude: Double?,
+    val altitude: Double?,
+    @SerializedName("fish_length") val fishLength: Double?,
```

Comment:
The '?' after the data type means it's a nullable type--the value could be something of that type, or it could be null.  I realized that the optional fields should be stored in nullable variables so the value can be null if the user doesn't enter anything (and then the keys will be omitted during json serialization)

Reply:
(Without the ? after the data types, they can't hold a null value)